### PR TITLE
Chore/notice banner for auth smtp changes

### PR DESCRIPTION
**Pull request by joshenlim**
- Set up notice banner for auth SMTP changes 
  - only shows up if custom smtp has not been set up
  - dismissible on a per project basis
- Deprecate old notice banner stuffs to do with ipv6, pgbouncer etc etc

Relevant discussion: https://github.com/orgs/supabase/discussions/29370

image

### CODE REVIEW

File: apps/studio/components/layouts/AppLayout/NoticeBanner.tsx [hunk 1]
```diff
@@ -1,124 +1,65 @@
...
-    ['projects', projectRef, 'pgbouncer-enabled'],
-    async ({ signal }) => {
-      let query = supabase.rpc('ipv6_active_status', { project_ref: projectRef }).returns<
-        {
-          pgbouncer_active: boolean
-          vercel_active: boolean
-        }[]
-      >()
+  const { data: authConfig } = useAuthConfigQuery({ projectRef })
+  const smtpEnabled = isSmtpEnabled(authConfig)
```
Comment (alaister) [inline apps/studio/components/layouts/AppLayout/NoticeBanner.tsx:30]: We should also take into account if they are using the auth hook here. So either smtp is enabled OR they have the auth hook configured we should not show the banner

**Review by alaister — COMMENTED** [review]
Looking good! Couple of comments:

File: apps/studio/components/interfaces/App/AppBannerWrapper.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ const AppBannerWrapper = ({ children }: PropsWithChildren<{}>) => {
     <div className="min-h-full flex flex-col">
       <div className="flex-none">
         {ongoingIncident && <IncidentBanner />}
-        {showNoticeBanner && <NoticeBanner />}
+        {true && <NoticeBanner />}
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/interfaces/App/AppBannerWrapper.tsx:33]: oh whoops! this should be the flag sorry 🙏

File: apps/studio/components/layouts/AppLayout/NoticeBanner.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/layouts/AppLayout/NoticeBanner.tsx:30]: this is specifically just the email hook right?

**Review by alaister — APPROVED** [review]
LGTM!

- Tested:
  - Project without anything configured
  - With smtp configured
  - With auth hook configured

**Pull request merged**